File: src/main/java/com/csl/guess/GameFor.java
package com.csl.guess;

import java.util.Random;
import java.util.Scanner;

public class GameFor {
    public static void main(String[] args) {

        int sum=0;
        for (int i=1; i<=10; i++) {
            sum+= i;
            System.out.println(sum);
        }

        Random random = new Random();
        int secret = random.nextInt(10)+1;
        System.out.println("secret number is: "+ secret);

        Scanner scanner = new Scanner(System.in);
        for(int i=1; i<=4; i++) {
            System.out.println("Please, enter a number("+ i+ "/4)");
            int number = scanner.nextInt();
            System.out.println("第"+ i+ "次 :"+number );

            if (number > secret) {
                System.out.println("lower");
            } else if (number < secret) {
                System.out.println("higher");
            } else {
                System.out.println("your are correct, the secret number is "+ secret);
                break;
            }
        }

        System.out.println("Finish");
    }
}

File: src/main/kotlin/com/csl/guess/GameFor.kt
package com.csl.guess

import java.util.*

fun main(args : Array<String>) {


    //kotlin的for loop規則:
    // for (range描述)
    //https://kotlinlang.org/docs/reference/ranges.html

    println("i in 1..4")
    for (i in 1..4) {
        println(i)
    }
    println("i in 1 until 5")
    for (i in 1 until 5) {
        println(i)
    }
    println("i in 1..4 step 2")
    for (i in 1..4 step 2) {
        println(i)
    }
    println("i in 4 downTo 1")
    for (i in 4 downTo 1) {
        println(i)
    }
    println("i in 4 downTo 1 step 2")
    for (i in 4 downTo 1 step 2) {
        println(i)
    }

    val scanner:Scanner = Scanner(System.`in`)
    val random:Random = Random()
    var secret:Int = random.nextInt(10)+1
    println("secret number is "+ secret)

    for (i in 1..4) {
        var number:Int = scanner.nextInt()

        if (number > secret)
            print("lower")
        else if (number < secret)
            println("higher")
        else {
            println("you are correct, the secret is "+ secret)
            break;
        }
    }

    println("Finish")


}